internal/internal_public.go

```diff
@@ -42,6 +45,18 @@ type (
 		HasNextPage() bool
 	}
 
+	// WorkflowExecutionContext represents one instance of workflow execution state in memory. Lock must be obtained before
+	// calling into any methods.
+	WorkflowExecutionContext interface {
+		sync.Locker
+		ProcessWorkflowTask(task *s.PollForDecisionTaskResponse, historyIterator HistoryIterator) (completeRequest interface{}, err error)
+		ProcessLocalActivityResult(lar *localActivityResult) (interface{}, error)
+		CompleteDecisionTask(waitLocalActivity bool) interface{}
+		GetDelayReportDuration(startTime time.Time) time.Duration
```

Comment:
Document semantic of the GetDelayReportDuration

Reply:
When i put doc for this method, I think I would rather expose a different method, say GetDecisionTimeout() and then it is the poller's decision to decide when to respond.